controllers/machine_controller_phases.go

```diff
@@ -187,7 +182,23 @@ func (r *MachineReconciler) reconcileBootstrap(ctx context.Context, cluster *clu
 	ready, err := external.IsReady(bootstrapConfig)
 	if err != nil {
 		return err
-	} else if !ready {
+	}
+
+	// Report a summary of current status of the bootstrap object defined for this machine.
+	conditions.SetMirror(m, clusterv1.BootstrapReadyCondition,
+		conditions.UnstructuredGetter(bootstrapConfig),
+		conditions.WithFallbackValue(ready, clusterv1.WaitingForDataSecretFallbackReason, clusterv1.ConditionSeverityInfo, ""),
+	)
+
+	// If the bootstrap data is populated, set ready and return.
+	if m.Spec.Bootstrap.DataSecretName != nil {
+		m.Status.BootstrapReady = true
+		conditions.MarkTrue(m, clusterv1.BootstrapReadyCondition)
+		return nil
+	}
```

Comment:
I'm wondering if this should be moved above the check for `m.Spec.Bootstrap.ConfigRef == nil`, currently it looks like we would never get far enough to mark `m.Status.BootstrapReady` or `clusterv1.BootstrapReadyCondition` to `True` in the case that a user provided the `DataSecretName` without providing the optional ConfigRef.

I don't think it affects any logic, but it seems like an omission that we haven't been setting this to date.

Reply:
@detiber now the use case when there is a user-provided bootstrap secret should be properly addressed, PTAL